array_methods_lab.js

```diff
@@ -0,0 +1,341 @@
+// # Exercises
+//
+console.log(`1. Given an array of numbers, write a function that returns
+  a new array where every element is incremented by 10.`);
+//
+// ```
+let inTenOut = [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+let byTen = inTenOut.map(elem => elem +=10);
+console.log(byTen);
+
+console.log(`\n2. Given an array, return a new array where every
+  non-string element is converted to an empty string.`);
+//
+// ```
+let strOrNot = ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+let testing = strOrNot.map(el => {
+  if (typeof el != 'string'){
+    return el ='';
+  } else {
+    return el;
+  }
+});
+//logging the new array
+ console.log(testing);
+
+
+console.log(`\n3. Write a function that returns only the even
+  elements in an array of numbers.`);
+//
+// ```
+let eveOut = [1,2,3,4,5,10,20]
+// Output: [2,4]
+// ```
+
+const evenFunc =(evenNum) =>{
+return evenNum.filter(el => el % 2 === 0);
+}
+console.log(evenFunc(eveOut));
+
+console.log(`\n4. Write a function that checks if all elements in the array are the same.`);
+//
+// ```
+let sameArr = [1,1,1,1,2];
+// Output: true
+//
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+const multfunc =(same) =>{
+  let num = same[0]
+  let test2 = same.every(el => {
+    if (el === num){
+      // console.log(el);
+      return true;
+    }
+      //console.log(el);
+      return false;
+  });
+  return test2;
+};
+console.log(multfunc(sameArr));
+
+console.log(`\n5. Write a function that returns the sum of every element in an array.`);
+//
+// ```
+let sumArr = [1,2,3,4,5];
+// Output: 15
+// ```
+const sumFunc = (sumIng) => {
+  let accArr = 0;
+  let sum = sumIng.reduce((acc,el) => {return acc + el}, accArr);
+
+  return sum;
+};
+
+console.log(sumFunc(sumArr));
+
+console.log(`\n6. Write a function that finds largest number in an array.`);
+//
+// ```
+let largestArr = [1,2,6,3,4,5];
+// Output: [6]
+// ```
+const largest = (larparam) => {
+ let large = largestArr.reduce((acc, currEl) => {
+    if (acc > currEl) {
+      return acc;
+    } else {
+      return currEl;
+    }
+  });
+  return large;
+};
+console.log(largest(largestArr));
+
+console.log(`\n7. Write a function 'countZeroes', which takes an array of numbers
+  as its argument and returns the amount of zeroes that occur in it.`);
+//
+// ```
+let zeroD30 = [1,0,0,2,0,3,0,0,0,4,7,0,100,8,0];
+// Output: 6
+// ```
+const countZeroes = (count0) => {
+  let count =0;
+  zeroD30.forEach(el => { if (el === 0){ count ++} });
+  return count;
+}
+
+console.log(countZeroes(zeroD30));
+
+console.log(`\n8. Write a function that doubles every element in an array.`);
+let double = [1,2,3,4,16];
+// Output: [2,4,6,8]
+// ```
+const doubleDutch = (dubNum) => {
+  let result = dubNum.map(el => el * 2)
+  return  result;
+}
+console.log(doubleDutch(double));
```

Comment:
Great ! To simplify even more return result in line 120 rather than creating a variable then returning it later on.